### PR TITLE
[docs] Improve Filtering page

### PR DESCRIPTION
**Pull request by m4theushw**
Fixes the documentation part in https://github.com/mui-org/material-ui-x/issues/1118#issuecomment-819500943.
Preview: https://deploy-preview-1671--material-ui-x.netlify.app/components/data-grid/filtering/

### CODE REVIEW

File: docs/src/pages/components/data-grid/filtering/filtering.md [hunk 1]
```diff
@@ -5,32 +5,36 @@ components: DataGrid, XGrid
...
 
-The type of the column is used for adapting the filtering. You can find the different supported types in the [columns section](/components/data-grid/columns/#column-types).
+The following demo allows to explore the different operators available:
 
 {{"demo": "pages/components/data-grid/filtering/FilterOperators.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### Disable filter
+## Disable filtering
 
 #### Globally
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/filtering/filtering.md:39]: h2 > h4 is not a valid HTML structure
```suggestion
**Globally**
```

x2

File: docs/src/pages/components/data-grid/filtering/filtering.md [hunk 2]
```diff
@@ -5,32 +5,36 @@ components: DataGrid, XGrid
...
+```
 
-### Toolbar
+{{"demo": "pages/components/data-grid/filtering/BasicFilteringGrid.js", "bg": "inline", "defaultCodeOpen": false}}
 
-In addition to the column menu that allows users to apply a filter, you can also show a toolbar:
+## Native filters
 
-{{"demo": "pages/components/data-grid/filtering/BasicToolbarFilteringGrid.js", "bg": "inline"}}
+A filter is composed of three parts: the column to filter, the value to look for and an operator (e.g. _contains_, _is before_, _is after_, etc.). The operator determines if a candidate value should be considered as a result. The candidate value used by the operator is the one corresponding to the `field` attribute or the `valueGetter` of the `GridColDef`. As part of the native column types, a set of operators is available. You can find the supported column types in the [columns section](/components/data-grid/columns/#column-types).
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/filtering/filtering.md:29]: Best to break line to make it easier to review:

```suggestion
A filter is composed of three parts: the column to filter, the value to look for, and an operator (e.g. _contains_, _is before_, _is after_, etc.).
The operator determines if a candidate value should be considered as a result.
The candidate value used by the operator is the one corresponding to the `field` attribute or the `valueGetter` of the `GridColDef`.
As part of the native column types, a set of operators is available.
You can find the supported column types in the [columns section](/components/data-grid/columns/#column-types).
```

File: docs/src/pages/components/data-grid/filtering/filtering.md [hunk 3]
```diff
@@ -93,15 +126,15 @@ Filtering can be run server-side by setting the `filterMode` prop to `server`, a
 />
 ```
 
-Below is very simple demo on how you could achieve server side filtering.
+Below is a very simple demo on how you could achieve server side filtering.
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/filtering/filtering.md:129]:
```suggestion
Below is a very simple demo on how you could achieve server-side filtering.
```

File: docs/src/pages/components/data-grid/filtering/filtering.md [hunk 4]
```diff
@@ -50,36 +54,65 @@ const columns = [{ field: 'image', filterable: false }];
...
+  InputComponent: RatingInputValue,
+  InputComponentProps: { type: 'number' },
+}
+```
 
-In this demo you will see how to create a complete new operator for a specific column.
+**Note**: If the column has a [`valueGetter`](/components/data-grid/columns/#value-getter), then `params.value` will be the resolved value.
 
-The rating column contains a new `From` operator, as you can see in the filter panel.
+In this demo you can see how to create a complete new operator for the Rating column.
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/filtering/filtering.md:111]:
```suggestion
In this demo, you can see how to create a completely new operator for the Rating column.
```

File: docs/src/pages/components/data-grid/filtering/filtering.md [hunk 5]
```diff
@@ -5,32 +5,36 @@ components: DataGrid, XGrid
...
+    items: [{ columnField: 'commodity', operatorValue: 'contains', value: 'rice' }],
+  }}
+/>
+```
 
-### Toolbar
+{{"demo": "pages/components/data-grid/filtering/BasicFilteringGrid.js", "bg": "inline", "defaultCodeOpen": false}}
 
-In addition to the column menu that allows users to apply a filter, you can also show a toolbar:
+## Native filters
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/filtering/filtering.md:27]: We use this terminology in https://deploy-preview-1671--material-ui-x.netlify.app/components/data-grid/columns/#column-types. I personally find it less confusing, I wonder what the others think.

```suggestion
## Predefined filters
```

File: docs/src/pages/components/data-grid/filtering/filtering.md [hunk 6]
```diff
@@ -93,15 +126,15 @@ Filtering can be run server-side by setting the `filterMode` prop to `server`, a
 />
 ```
 
-Below is very simple demo on how you could achieve server side filtering.
+Below is a very simple demo on how you could achieve server side filtering.
 
 {{"demo": "pages/components/data-grid/filtering/ServerFilterGrid.js", "bg": "inline"}}
 
-### Controlled filtering
+<!-- ## Controlled filtering -->
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/filtering/filtering.md:133]: I would have personally kept the header up and clarify the description (WIP is about the docs, not the implementation), but the current approach can work too.

**Review by oliviertassinari — COMMENTED** [review]
👍

File: docs/src/pages/components/data-grid/filtering/filtering.md [hunk 7]
```diff
@@ -50,36 +54,65 @@ const columns = [{ field: 'image', filterable: false }];
...
+On most column types, a text field is used. However, a custom component can be rendered instead.
 
-The rating column reuses the numeric operator, but the input value is a new rating component.
+In this demo the Rating column reuses the numeric filter and the same rating component is used to the enter the value of the filter.
 
-{{"demo": "pages/components/data-grid/filtering/ExtendNumericOperator.js", "bg": "inline"}}
+{{"demo": "pages/components/data-grid/filtering/ExtendNumericOperator.js", "bg": "inline", "defaultCodeOpen": false}}
 
-2. **Custom column type**.
+### Reuse filter operators on a custom column type
```
Comment (dtassone) [inline docs/src/pages/components/data-grid/filtering/filtering.md:71]: that's quite a long title

File: docs/src/pages/components/data-grid/filtering/filtering.md (hunk 6 above)
Comment (dtassone) [reply docs/src/pages/components/data-grid/filtering/filtering.md:133]: I want to look into this controlled state and filters this week

File: docs/src/pages/components/data-grid/filtering/filtering.md (hunk 5 above)
Comment (m4theushw) [reply docs/src/pages/components/data-grid/filtering/filtering.md]: I changed to "predefined". The page you linked uses both terms. We should keep just one. I'll fix it in another PR to improve the Columns page.

image

File: docs/src/pages/components/data-grid/filtering/filtering.md (hunk 7 above)
Comment (m4theushw) [reply docs/src/pages/components/data-grid/filtering/filtering.md]: Changed to "Reuse filter operators".

Comment (dtassone) [reply docs/src/pages/components/data-grid/filtering/filtering.md]: Extend?

**Review by dtassone — APPROVED** [review]

File: docs/src/pages/components/data-grid/filtering/filtering.md (hunk 5 above)
Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/filtering/filtering.md]: The usage of "native" here might be used as a synonym, to avoid the repetition.

**Pull request merged**